app/components/channel_drawer/channels_list/list/index.js

```diff
@@ -34,6 +34,7 @@ function mapStateToProps(state) {
 
     return {
         canCreatePrivateChannels: showCreateOption(state, config, license, currentTeamId, General.PRIVATE_CHANNEL, isAdmin, isSystemAdmin),
+        canCreatePublicChannels: showCreateOption(state, config, license, currentTeamId, General.OPEN_CHANNEL, isAdmin, isSystemAdmin),
```

Comment:
We really need to look at creating some selectors for this at some point where you just provide the necessary fields instead of passing in everything needed manually